crates/precompile/src/lib.rs

```diff
@@ -191,6 +192,22 @@ impl Precompiles {
         })
     }
 
+    pub fn cancun() -> &'static Self {
+        static INSTANCE: OnceCell<Precompiles> = OnceCell::new();
+        INSTANCE.get_or_init(|| {
+            let mut precompiles = Self::berlin().clone();
```

Comment:
for precompiles, we should consider either an enum

```rust
enum Precompile {
  Fun(fn(&[u8], u64) -> PrecompileResult),
  Fn(Arc<dyn Send + Sync + Fn(&[u8], u64) -> PrecompileResult)
}
```

or always use the Arc variant, 

and a static mut Precompile list that supports configuring global defaults

Reply:
This is in my TODO list :)

Reply:
hm we can simplify a lot of things about initialization by adding function signature that has `CfgEnv` input
No global initialization at all, and we set `KzgSettings` inside `CfgEnv`.

@DaniPopes @mattsse @Rjected wdyt?

Reply:
> No global initialization at all, and we set KzgSettings inside CfgEnv.

I am open to that, as long as we can set a default that's hard-coded to Cancun/Sepolia/.. mainnet?

Reply:
> > No global initialization at all, and we set KzgSettings inside CfgEnv.
> 
> I am open to that, as long as we can set a default that's hard-coded to Cancun/Sepolia/.. mainnet?

yeah this seems much nicer than messing around with globals